File: C/Lista_04(Vetores)/Comerciante.c
#include <stdio.h>
#include <string.h>

void limpar_entrada() { 
    char c; 
    while ((c = getchar()) != '\n' && c != EOF) {} 
} 

void ler_texto(char *buffer, int length) { 
    fgets(buffer, length, stdin); 
    strtok(buffer, "\n"); 
}

int main() {

    int n;

    printf("Serao digitados dados de quantos produtos? ");
    scanf("%d", &n);

    char nome[n][50];
    double precoCompra[n];
    double precoVenda[n];

    for(int i=0;i<n;i++){
        printf("Produto %d:\n", i+1);
        printf("Nome: ");
        limpar_entrada();
        ler_texto(nome[i], 50);
        printf("Preco de compra: ");
        scanf("%lf", &precoCompra[i]);
        printf("Preco de venda: ");
        scanf("%lf", &precoVenda[i]);
    }

    int cont1=0, cont2=0, cont3=0;
    double totalCompra=0, totalVenda=0;

    for(int i=0;i<n;i++){
        if(precoVenda[i]-precoCompra[i] < precoCompra[i]/10){
            cont1++;
        }
        else if(precoVenda[i]-precoCompra[i] < precoCompra[i]/5){
            cont2++;
        }
        else{
            cont3++;
        }
        totalCompra += precoCompra[i];
        totalVenda += precoVenda[i];
    }

    printf("\nRELATORIO:\n");
    printf("Lucro abaixo de 10%%: %d\n", cont1);
    printf("Lucro entre 10%% e 20%%: %d\n", cont2);
    printf("Lucro acima de 20%%: %d\n", cont3);
    printf("Valor total de compra: %.2lf\n", totalCompra);
    printf("Valor total de venda: %.2lf\n", totalVenda);
    printf("Lucro total: %.2lf", (totalVenda-totalCompra));

    return 0;

}

File: Python/Lista_04(Vetores)/Comerciante.py
n = int(input("Serao digitados dados de quantos produtos? "))

nome = [0 for x in range(n)]
precoC = [0 for x in range(n)]
precoV = [0 for x in range(n)]

for i in range(0,n):
    print(f"Produto {i+1}:")
    nome[i] = input("Nome: ")
    precoC[i] = float(input("Preco de compra: "))
    precoV[i] = float(input("Preco de venda: "))

cont1 = 0
cont2 = 0
cont3 = 0

for i in range(0,n):
    if precoV[i]<precoC[i]*1.10:
        cont1+=1
    elif precoV[i]<=precoC[i]*1.20:
        cont2+=1
    else:
        cont3+=1

totC = 0.00
totV = 0.00

for i in range(0,n):
    totC+=precoC[i]
    totV+=precoV[i]

lucro = totV - totC

print("\nRELATORIO")
print(f"Lucro abaixo de 10%: {cont1}")
print(f"Lucro entre 10% e 20%: {cont2}")
print(f"Lucro acima de 20%: {cont3}")
print(f"Valor total de compra: {totC:.2f}")
print(f"Valor total de venda: {totV:.2f}")
print(f"Lucro: {lucro:.2f}")